Gems/AudioSystem/Code/Source/Engine/FileCacheManager.cpp

```diff
@@ -340,44 +358,49 @@ namespace Audio
 
 #if !defined(AUDIO_RELEASE)
     ///////////////////////////////////////////////////////////////////////////////////////////////
-    void CFileCacheManager::DrawDebugInfo(IRenderAuxGeom& auxGeom, const float posX, const float posY)
+    void CFileCacheManager::DrawDebugInfo(AzFramework::DebugDisplayRequests& debugDisplay, const float posX, const float posY)
     {
         if (CVars::s_debugDrawOptions.AreAllFlagsActive(DebugDraw::Options::FileCacheInfo))
         {
             const auto frameTime = AZStd::chrono::system_clock::now();
 
-            const float entryDrawSize = 1.5f;
+            const float entryDrawSize = 0.8f;
             const float entryStepSize = 15.0f;
             float positionY = posY + 20.0f;
             float positionX = posX + 20.0f;
             float time = 0.0f;
             float ratio = 0.0f;
             float originalAlpha = 0.7f;
-            float* color = nullptr;
 
             // The colors.
-            float white[4] = { 1.0f, 1.0f, 1.0f, originalAlpha };
-            float cyan[4] = { 0.0f, 1.0f, 1.0f, originalAlpha };
-            float orange[4] = { 1.0f, 0.5f, 0.0f, originalAlpha };
-            float green[4] = { 0.0f, 1.0f, 0.0f, originalAlpha };
-            float red[4] = { 1.0f, 0.0f, 0.0f, originalAlpha };
-            float redish[4] = { 0.7f, 0.0f, 0.0f, originalAlpha };
-            float blue[4] = { 0.1f, 0.2f, 0.8f, originalAlpha };
-            float yellow[4] = { 1.0f, 1.0f, 0.0f, originalAlpha };
-            float darkish[4] = { 0.3f, 0.3f, 0.3f, originalAlpha };
-
-            auxGeom.Draw2dLabel(posX, positionY, 1.6f, orange, false,
-                "FileCacheManager (%zu of %zu KiB) [Entries: %zu]", m_currentByteTotal >> 10, m_maxByteTotal >> 10, m_audioFileEntries.size());
-            positionY += 15.0f;
+            AZ::Color white{ 1.0f, 1.0f, 1.0f, originalAlpha };     // file is use-counted
+            AZ::Color cyan{ 0.0f, 1.0f, 1.0f, originalAlpha };      // file is global scope
+            AZ::Color orange{ 1.0f, 0.5f, 0.0f, originalAlpha };    // header color
+            AZ::Color green{ 0.0f, 1.0f, 0.0f, originalAlpha };     // file is removable
+            AZ::Color red{ 1.0f, 0.0f, 0.0f, originalAlpha };       // memory allocation failed
+            AZ::Color redish{ 0.7f, 0.0f, 0.0f, originalAlpha };    // file not found
+            AZ::Color blue{ 0.1f, 0.2f, 0.8f, originalAlpha };      // file is loading
+            AZ::Color yellow{ 1.0f, 1.0f, 0.0f, originalAlpha };    // file is level scope
+            AZ::Color darkish{ 0.3f, 0.3f, 0.3f, originalAlpha };   // file is not loaded
```

Comment:
Debug draw constants like these can be read from the SettingsRegistry to allow data driven configuration.

Reply:
Good idea, but this can be addressed later.